tests/eas/cpuhp.py

```diff
@@ -168,7 +142,13 @@ def _random_cpuhp_script(cls):
             if sleep_max > 0:
                 sleep_dur_sec = random.randint(sleep_min, sleep_max)/1000.0
                 script.append(shift + 'sleep {}'.format(sleep_dur_sec))
-        script.append('done')
+        if timeout:
```

Comment:
`timeout`'s a mandatory parameter now, so could this be unified?
That way, maybe you could just used a time-based loop like this:
```bash
script.append('end=$((SECONDS+{}))'.format(timeout)
script.append('while [ $SECONDS -lt $end ]; do')
# Insert hotplug action here
script.append('done')
```
You could also use `date` instead (Stackoverflow is a wonderful place https://stackoverflow.com/a/11198713/5096023).

Reply:
The time-based loop won't work well because the number of actions in the loop is arbitrary long, so one iteration of the loop can be arbitrary long as well. That why I kill it 'from the outside'.

But yeah I can remove the `if timeout:` part, n/p.